src/main/java/teammates/logic/core/SendgridService.java

```diff
@@ -37,9 +36,10 @@ public Sendgrid parseToEmail(EmailWrapper wrapper) throws JSONException {
      * {@inheritDoc}
      */
     @Override
-    public void sendEmail(EmailWrapper wrapper) throws JSONException, IOException {
-        Sendgrid email = parseToEmail(wrapper);
-        email.send();
+    public void sendEmail(EmailWrapper wrapper) throws SendGridException {
+        Email email = parseToEmail(wrapper);
+        SendGrid sendgrid = new SendGrid(Config.SENDGRID_APIKEY);
+        sendgrid.send(email);
```

Comment:
We could, in fact Mailgun and Mailjet also return response codes with their email sending function. JavaMail is beaten quite badly in this department.